File: Sources/Klck/Model/Setlist.swift
import Foundation

/// One stop in a setlist: a reference to a saved preset, with an optional
/// auto-advance after a number of bars (0 = advance manually).
struct SetlistItem: Identifiable, Codable, Equatable {
    var id: UUID = UUID()
    var presetID: UUID
    var advanceAfterBars: Int = 0
}

/// An ordered chain of presets for a song, gig, or practice routine.
struct Setlist: Identifiable, Codable, Equatable {
    var id: UUID = UUID()
    var name: String
    var items: [SetlistItem] = []
}

File: android/app/src/main/java/com/klck/metronome/model/Setlist.kt
package com.klck.metronome.model

import kotlinx.serialization.Serializable
import java.util.UUID

/** One stop in a setlist. Mirrors Sources/Klck/Model/Setlist.swift::SetlistItem. */
@Serializable
data class SetlistItem(
    @Serializable(with = UuidSerializer::class)
    val id: UUID = UUID.randomUUID(),
    @Serializable(with = UuidSerializer::class)
    val presetID: UUID,
    val advanceAfterBars: Int = 0,   // 0 = manual advance
)

/** An ordered chain of presets. Mirrors Sources/Klck/Model/Setlist.swift::Setlist. */
@Serializable
data class Setlist(
    @Serializable(with = UuidSerializer::class)
    val id: UUID = UUID.randomUUID(),
    val name: String,
    val items: List<SetlistItem> = emptyList(),
)
